security/advancedtls/crl.go

```diff
@@ -192,25 +188,25 @@ func x509NameHash(r pkix.RDNSequence) string {
 	return fmt.Sprintf("%08x", fileHash)
 }
 
-// CheckRevocation checks the connection for revoked certificates based on RFC5280.
+// checkRevocation checks the connection for revoked certificates based on RFC5280.
 // This implementation has the following major limitations:
 //   - Indirect CRL files are not supported.
 //   - CRL loading is only supported from directories in the X509_LOOKUP_hash_dir format.
 //   - OnlySomeReasons is not supported.
 //   - Delta CRL files are not supported.
 //   - Certificate CRLDistributionPoint must be URLs, but are then ignored and converted into a file path.
 //   - CRL checks are done after path building, which goes against RFC4158.
-func CheckRevocation(conn tls.ConnectionState, cfg RevocationConfig) error {
-	return CheckChainRevocation(conn.VerifiedChains, cfg)
+func checkRevocation(conn tls.ConnectionState, cfg RevocationConfig) error {
+	return checkChainRevocation(conn.VerifiedChains, cfg)
 }
 
-// CheckChainRevocation checks the verified certificate chain
+// checkChainRevocation checks the verified certificate chain
 // for revoked certificates based on RFC5280.
-func CheckChainRevocation(verifiedChains [][]*x509.Certificate, cfg RevocationConfig) error {
+func checkChainRevocation(verifiedChains [][]*x509.Certificate, cfg RevocationConfig) error {
```

Comment:
Thinking about the target state - aren't we going to provide some interface that will allow users to define such functions? If yes, I think it's better to keep this one public, or maybe provide a builder (in this case private is fine).

Reply:
I don't think so - the interface is the CRLProvider, so the user is just responsible for giving us CRLs. We don't provide any overrides for the actual revocation check